docs/docs/slot-validation-actions.mdx

```diff
@@ -0,0 +1,103 @@
+---
+id: slot-validation-actions
+sidebar_label: Slot Validation Actions
+title: Slot Validation Actions
+abstract: A slot validation action is a special type of custom action, designed to handle custom extraction and/or validation of slot values. This can be used to validate slots with predefined mappings or extract slots with custom mappings.
+---
+
+Rasa SDK provides three base classes that you can inherit from in order to implement custom slot extraction and
+validation logic:
+
+## `Action` class
+
+You can use this class to create a regular [custom action](./custom-actions.mdx). The base class will not prevent you
+from returning arbitrary events, however a slot validation action should only return `SlotSet` and `BotUttered` events.
+Any other event type will be filtered out by the default action [`action_extract_slots`](./default-actions.mdx#action_extract_slots).
+The name of the custom action must be specified in the `action` key of the relevant
+[custom slot mapping](./domain.mdx#custom-slot-mappings) in the domain.
+Note that the action name must be listed in the domain `actions` section too.
+
+## `ValidationAction` class
+
+You can extend the `ValidationAction` class in the Rasa SDK to define custom extraction and / or validation of slots that
+can be set or updated outside of a form context.
+With this option, you do not need to specify the `action` key in the [custom slot mapping](./domain.mdx#custom-slot-mappings).
+
+Note that you do not need to implement the `name` method in the custom action extending `ValidationAction`, since this
+is already provided: `action_validate_slot_mappings`. However this name must be added to the domain `actions` list.
+If you override the `name` method, the custom validation action will not run because the original name is hardcoded
+in the call that the default action `action_extract_slots` makes to the action server.
+
```

Comment:
What happens with non-rasa-sdk action servers here? Can the users implement their own `action_validate_slot_mappings` ?

Reply:
This pertains to the last point in the PR about updating action server docs - Some of this might need to be moved over there entirely, so that people with non-sdk action servers also can refer to and understand how to use custom actions for slot mappings.

Reply:
> What happens with non-rasa-sdk action servers here?

Completely forgot this is also possible 🙈 I guess they'll have to implement their own `ValidationAction` mirroring the Python sdk. Should I then open a new issue for action server docs update? What kind of issues need to be updated, I made a note about knowledge base slots code that might need to be updated with  mapping of `type: custom`?

Reply:
>Completely forgot this is also possible

And I completely neglected to consider it in this review up till now, so equally guilty 🙈

Yes, let's make a new issue for sdk docs updates. 
That issue can include 
- deciding how much of the slot validation action docs should move to the action server docs
- describing the behaviour of rasa-sdk's `ValidationAction` in broad strokes to allow other language implementations of it.

Reply:
In our inbox: https://github.com/RasaHQ/rasa-sdk/issues/597